### DISCUSSION TITLE
IL Analysis Passes

### DISCUSSION THREAD

**Discussion by toolCHAINZ** [Vector35/binaryninja-api#1573] [General] [answered]
Does Binary Ninja support user-defined "analysis passes" for altering the lifting from ASM => LLIL => MLIL => HLIL?

**Comment by psifertex** [comment, marked as answer]
No, but this is definitely on the road-map. At one point we thought that feature might be coming with HLIL, but since HLIL is landing on dev later today, that's obviously not the case. :-)

It's still high on our priority list but we don't have any ETA for it as it's a fairly large refactor.

**Reply by plafosse** [reply]
This issue will track this feature: #1606

**Reply by psifertex** [reply]
Wow, no clue how I missed that we didn't have this as a bug yet, but we do now! See #1606
